### DISCUSSION TITLE
Localization of App

### DISCUSSION THREAD

**Discussion by pczern** [a-h/templ#980] [Q&A] [answered]
Hey I have been creating templ layouts for my app and need to set the `langCode` and add localization. I have seen the example with https://github.com/invopop/ctxi18n but I am not sure if it's still maintained and still the recommendation. It seems there is also https://github.com/nicksnyder/go-i18n which has more maintainers. As I am still relatively new to Go and its ecosystem I'd love to know what's the best path to take.

**Comment by pharrisee** [comment, marked as answer]
We've used the invopop one in a small project in work with great success.  I think the library does it's job so there's not much to maintain tbh.  The 2 issues there are are 1 enhancement for machine translation and 1 other which also might turn into an enhancement, so it seems rock steady to me.

We took a good look at the nicksnyder library some time back and whilst it was maybe more capable, it was also very much more complicated to get up and running. (but that was some time ago)

**Reply by pczern** [reply]
Thank you, I just checked the nicksnyder one out and I think it has a very long syntax for translating. 
I will also go with [ctxi18n](https://github.com/invopop/ctxi18n) as the method names are much shorter and thus much better readable in the templates.
